agent/src/util/device_plugin_service.rs

```diff
@@ -58,12 +63,27 @@ pub struct InstanceInfo {
     pub list_and_watch_message_sender: broadcast::Sender<ListAndWatchMessageKind>,
     /// Instance's `InstanceConnectivityStatus`
     pub connectivity_status: InstanceConnectivityStatus,
+    /// Instance's hash id
+    pub instance_id: String,
+    /// Device that the instance represents.
+    /// Contains information about environment variables and volumes that should be mounted
+    /// into requesting Pods.
+    pub device: Device,
+}
+
+#[derive(Clone, Debug, Default)]
+pub struct InstanceConfig {
+    /// Sender to tell Configuration device plugin `list_and_watch` to either prematurely continue looping or end
+    pub usage_update_message_sender: Option<broadcast::Sender<ListAndWatchMessageKind>>,
+    /// Map of all Instances from the same Configuration
+    pub instances: HashMap<String, InstanceInfo>,
 }
 
-pub type InstanceMap = Arc<RwLock<HashMap<String, InstanceInfo>>>;
+pub type InstanceMap = Arc<RwLock<InstanceConfig>>;
```

Comment:
InstanceMap should probably be renamed, it is more than a map at this point.

Reply:
rename.